hs_core/hydroshare/resource.py

```diff
@@ -686,7 +690,18 @@ def delete_resource(pk):
 
     Note:  Only HydroShare administrators will be able to delete formally published resour
     """
-    utils.get_resource_by_shortkey(pk).delete()
+    #utils.get_resource_by_shortkey(pk).delete()
+    res = utils.get_resource_by_shortkey(pk)
+    for fl in res.files.all():
+        fl.resource_file.delete()
+
+    for bag in res.bags.all():
+        bag.bag.delete()
+        bag.delete()
+
+    res.metadata.delete_all_elements()
+    # deleting the metadata object deletes the resource also
+    res.metadata.delete()
```

Comment:
If you want custom delete behavior, which is what clean up of associated files is, you need to implement a delete() method on the model class.